src/datasources/db/v2/postgres-database.service.spec.ts

```diff
@@ -0,0 +1,169 @@
+        PostgresDatabaseService,
+      ],
+    }).compile();
+
+    postgresqlService = moduleRef.get<PostgresDatabaseService>(
+      PostgresDatabaseService,
+    );
+  });
+
+  afterAll(async () => {
+    await postgresqlService.destroyDatabaseConnection();
+    await moduleRef.close();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('getDataSource()', () => {
+    it('Should return the datasource', () => {
+      const result = postgresqlService.getDataSource();
+
+      expect(result).toBeInstanceOf(DataSource);
+    });
+  });
+
+  describe('isInitialized()', () => {
+    it('Should return true if the datasource has been initialized', async () => {
+      await postgresqlService.initializeDatabaseConnection();
+
+      const isInitialized = postgresqlService.isInitialized();
+
+      expect(isInitialized).toBe(true);
+    });
+
+    it('Should return false if the data source has not been initialized', async () => {
+      await postgresqlService.destroyDatabaseConnection();
+      const isInitialized = postgresqlService.isInitialized();
+      await postgresqlService.initializeDatabaseConnection();
+
+      expect(isInitialized).toBe(false);
+    });
+  });
+
+  describe('initializeDatabaseConnection()', () => {
+    it('Should initialize the data source if not initialized', async () => {
+      await postgresqlService.destroyDatabaseConnection();
+      const isPrematurelyInitialized = postgresqlService.isInitialized();
+      await postgresqlService.initializeDatabaseConnection();
+
+      const isInitialized = postgresqlService.isInitialized();
+
+      expect(isPrematurelyInitialized).toBe(false);
+      expect(isInitialized).toBe(true);
+    });
+
+    it('Should return the data source if already initialized', async () => {
+      const result = await postgresqlService.initializeDatabaseConnection();
+
+      expect(result.query).not.toBeUndefined();
+    });
+  });
+
+  describe('destroyDatabaseConnection()', () => {
+    it('Should destroy the data source if initialized', async () => {
+      await postgresqlService.initializeDatabaseConnection();
+      const isInitialized = postgresqlService.isInitialized();
+
+      await postgresqlService.destroyDatabaseConnection();
+      const isDatasourceAlive = postgresqlService.isInitialized();
+
+      expect(isInitialized).toBe(true);
+      expect(isDatasourceAlive).toBe(false);
+    });
+  });
+
+  describe('getRepository()', () => {
+    class MockEntity {}
+
+    it('Should return a repository for the given entity', async () => {
+      const mockRepository = {} as Promise<Repository<ObjectLiteral>>;
+      jest
+        .spyOn(postgresqlService, 'getRepository')
+        .mockReturnValue(mockRepository);
+
+      const result =
+        await postgresqlService.getRepository<MockEntity>(MockEntity);
+
+      expect(postgresqlService.getRepository).toHaveBeenCalledWith(MockEntity);
+      expect(result).toBe(mockRepository);
+    });
```

Comment:
Is this test necessary? It's essentially testing `mockReturnValue`.

Reply:
The purpose of this test is to ensure that for a given entity, we retrieve the correct repository. Do you think we should refactor or remove it?

Reply:
If it's integral, we can keep it. On face value, it's a bit confusing as you mock a function and test that function directly. Could you add a comment?

Reply:
I'll proceed with deleting the test if that works for you. @iamacook @hectorgomezv

Reply:
Removed
https://github.com/safe-global/safe-client-gateway/pull/1988/commits/30668354922839bc2e4ecb06d5f93bbd0ce92a15